staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go

```diff
@@ -193,6 +200,63 @@ func (c *threadSafeMap) ByIndex(indexName, indexedValue string) ([]interface{},
 	return list, nil
 }
 
+// ByIndexes Joint indexing by multiple indexers while supporting Operator
+func (c *threadSafeMap) ByIndexes(conds IndexConditions, opts *JointIndexOptions) ([]interface{}, error) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	for i, cond := range conds {
+		populatedIndex, err := c.storeKeysByIndexCondition(cond)
+		if err != nil {
+			return nil, err
+		}
+		conds[i].indexedResult = populatedIndex
```

Comment:
Here it will race write, if called concurrently ByIndex, also I think it is not a good behavior to modify the args